### PR TITLE
Ignore changes to Status in consul_config_entry for http-route, tcp-route and api-gateway

### PR DESCRIPTION
**Pull request by remilapeyre**
Closes https://github.com/hashicorp/terraform-provider-consul/issues/359

Also update the Consul API client to v1.25.1.

### CODE REVIEW

File: consul/resource_consul_config_entry_ce_test.go [hunk 1]
```diff
@@ -841,3 +844,20 @@ resource "consul_config_entry" "jwt_provider" {
...
+
+	config_json = jsonencode({
+	  Hostnames = null
+	  Rules     = null
+
+	  Parents = [{
+		  Kind = "api-gateway"
+		  Name = "test"
+    }]
+	})
```
Comment (roncodingenthusiast) [inline consul/resource_consul_config_entry_ce_test.go:861]: nit: formatting looks a bit off here

File: go.mod [hunk 2]
```diff
@@ -5,14 +5,15 @@ require (
...
 	github.com/hashicorp/terraform-plugin-sdk v1.17.2
 	github.com/mitchellh/mapstructure v1.5.0
+	golang.org/x/exp v0.0.0-20230817173708-d852ddb80c63
 )
 
 require (
 	cloud.google.com/go/compute v1.18.0 // indirect
 	cloud.google.com/go/compute/metadata v0.2.3 // indirect
 	cloud.google.com/go/iam v0.12.0 // indirect
+	github.com/armon/go-metrics v0.4.1 // indirect
```
Comment (roncodingenthusiast) [inline go.mod:15]: nit: I'd suggest using `hashicorp/go-metrics` rather than Armon's go-metrics

**Review by roncodingenthusiast — APPROVED** [review]
looks good to me, minor nitpicks. I think we are moving most things to hashicorp/go-metrics in general as `armon/go-metrics` was migrated to that so they are identical

File: consul/resource_consul_config_entry_ce_test.go (hunk 1 above)
Comment (remilapeyre) [reply consul/resource_consul_config_entry_ce_test.go:861]: Thanks

**Pull request merged**